File: src/assets/python/autogen/base/model/load.py
{{#if (eq modelProvider "Bedrock")}}
import os
from autogen_ext.models.anthropic import AnthropicBedrockChatCompletionClient
from autogen_core.models import ModelInfo, ModelFamily

# Uses global inference profile for Claude Sonnet 4.5
# https://docs.aws.amazon.com/bedrock/latest/userguide/inference-profiles-support.html
MODEL_ID = "global.anthropic.claude-sonnet-4-5-20250929-v1:0"


def load_model() -> AnthropicBedrockChatCompletionClient:
    """Get Bedrock model client using IAM credentials."""
    return AnthropicBedrockChatCompletionClient(
        model=MODEL_ID,
        model_info=ModelInfo(
            vision=False,
            function_calling=True,
            json_output=False,
            family=ModelFamily.CLAUDE_4_SONNET,
            structured_output=True
        ),
        bedrock_info={"aws_region": os.environ.get("AWS_REGION", "us-east-1")}
    )
{{/if}}
{{#if (eq modelProvider "Anthropic")}}
import os
from autogen_ext.models.anthropic import AnthropicChatCompletionClient
from bedrock_agentcore.identity.auth import requires_api_key

IDENTITY_PROVIDER_NAME = "{{identityProviders.[0].name}}"
IDENTITY_ENV_VAR = "{{identityProviders.[0].envVarName}}"


@requires_api_key(provider_name=IDENTITY_PROVIDER_NAME)
def _agentcore_identity_api_key_provider(api_key: str) -> str:
    """Fetch API key from AgentCore Identity."""
    return api_key


def _get_api_key() -> str:
    """
    Uses AgentCore Identity for API key management in deployed environments.
    For local development, run via 'agentcore dev' which loads agentcore/.env.
    """
    if os.getenv("LOCAL_DEV") == "1":
        api_key = os.getenv(IDENTITY_ENV_VAR)
        if not api_key:
            raise RuntimeError(
                f"{IDENTITY_ENV_VAR} not found. Run via 'agentcore dev' to load agentcore/.env"
            )
        return api_key
    return _agentcore_identity_api_key_provider()


def load_model() -> AnthropicChatCompletionClient:
    """Get authenticated Anthropic model client."""
    return AnthropicChatCompletionClient(
        model="claude-sonnet-4-5-20250929",
        api_key=_get_api_key()
    )
{{/if}}
{{#if (eq modelProvider "OpenAI")}}
import os
from autogen_ext.models.openai import OpenAIChatCompletionClient
from bedrock_agentcore.identity.auth import requires_api_key

IDENTITY_PROVIDER_NAME = "{{identityProviders.[0].name}}"
IDENTITY_ENV_VAR = "{{identityProviders.[0].envVarName}}"


@requires_api_key(provider_name=IDENTITY_PROVIDER_NAME)
def _agentcore_identity_api_key_provider(api_key: str) -> str:
    """Fetch API key from AgentCore Identity."""
    return api_key


def _get_api_key() -> str:
    """
    Uses AgentCore Identity for API key management in deployed environments.
    For local development, run via 'agentcore dev' which loads agentcore/.env.
    """
    if os.getenv("LOCAL_DEV") == "1":
        api_key = os.getenv(IDENTITY_ENV_VAR)
        if not api_key:
            raise RuntimeError(
                f"{IDENTITY_ENV_VAR} not found. Run via 'agentcore dev' to load agentcore/.env"
            )
        return api_key
    return _agentcore_identity_api_key_provider()


def load_model() -> OpenAIChatCompletionClient:
    """Get authenticated OpenAI model client."""
    return OpenAIChatCompletionClient(
        model="gpt-4o",
        api_key=_get_api_key()
    )
{{/if}}
{{#if (eq modelProvider "Gemini")}}
import os
from autogen_ext.models.openai import OpenAIChatCompletionClient
from bedrock_agentcore.identity.auth import requires_api_key

IDENTITY_PROVIDER_NAME = "{{identityProviders.[0].name}}"
IDENTITY_ENV_VAR = "{{identityProviders.[0].envVarName}}"


@requires_api_key(provider_name=IDENTITY_PROVIDER_NAME)
def _agentcore_identity_api_key_provider(api_key: str) -> str:
    """Fetch API key from AgentCore Identity."""
    return api_key


def _get_api_key() -> str:
    """
    Uses AgentCore Identity for API key management in deployed environments.
    For local development, run via 'agentcore dev' which loads agentcore/.env.
    """
    if os.getenv("LOCAL_DEV") == "1":
        api_key = os.getenv(IDENTITY_ENV_VAR)
        if not api_key:
            raise RuntimeError(
                f"{IDENTITY_ENV_VAR} not found. Run via 'agentcore dev' to load agentcore/.env"
            )
        return api_key
    return _agentcore_identity_api_key_provider()


def load_model() -> OpenAIChatCompletionClient:
    """Get authenticated Gemini model client via OpenAI-compatible API."""
    return OpenAIChatCompletionClient(
        model="gemini-2.0-flash",
        api_key=_get_api_key(),
        base_url="https://generativelanguage.googleapis.com/v1beta/openai/"
    )
{{/if}}

File: src/assets/typescript/http/strands/base/model/load.ts
{{#if (eq modelProvider "Bedrock")}}
import { BedrockModel } from '@strands-agents/sdk/models/bedrock';

export function loadModel(): BedrockModel {
  return new BedrockModel({ modelId: 'global.anthropic.claude-sonnet-4-5-20250929-v1:0' });
}
{{/if}}
{{#if (eq modelProvider "Anthropic")}}
import { AnthropicModel } from '@strands-agents/sdk/models/anthropic';
import { withApiKey } from 'bedrock-agentcore/identity';

const IDENTITY_PROVIDER_NAME = '{{identityProviders.[0].name}}';
const IDENTITY_ENV_VAR = '{{identityProviders.[0].envVarName}}';

async function getApiKey(): Promise<string> {
  if (process.env.LOCAL_DEV === '1') {
    const apiKey = process.env[IDENTITY_ENV_VAR];
    if (!apiKey) {
      throw new Error(`${IDENTITY_ENV_VAR} not found. Add ${IDENTITY_ENV_VAR}=your-key to .env.local`);
    }
    return apiKey;
  }
  return withApiKey({ providerName: IDENTITY_PROVIDER_NAME }, async (apiKey: string) => apiKey)();
}

export function loadModel(): AnthropicModel {
  return new AnthropicModel({
    clientArgs: { apiKey: getApiKey },
    modelId: 'claude-sonnet-4-5-20250929',
    maxTokens: 5000,
  });
}
{{/if}}
{{#if (eq modelProvider "OpenAI")}}
import { OpenAIModel } from '@strands-agents/sdk/models/openai';
import { withApiKey } from 'bedrock-agentcore/identity';

const IDENTITY_PROVIDER_NAME = '{{identityProviders.[0].name}}';
const IDENTITY_ENV_VAR = '{{identityProviders.[0].envVarName}}';

async function getApiKey(): Promise<string> {
  if (process.env.LOCAL_DEV === '1') {
    const apiKey = process.env[IDENTITY_ENV_VAR];
    if (!apiKey) {
      throw new Error(`${IDENTITY_ENV_VAR} not found. Add ${IDENTITY_ENV_VAR}=your-key to .env.local`);
    }
    return apiKey;
  }
  return withApiKey({ providerName: IDENTITY_PROVIDER_NAME }, async (apiKey: string) => apiKey)();
}

export function loadModel(): OpenAIModel {
  return new OpenAIModel({
    clientArgs: { apiKey: getApiKey },
    modelId: 'gpt-4.1',
  });
}
{{/if}}
{{#if (eq modelProvider "Gemini")}}
import { GoogleModel } from '@strands-agents/sdk/models/google';
import { withApiKey } from 'bedrock-agentcore/identity';

const IDENTITY_PROVIDER_NAME = '{{identityProviders.[0].name}}';
const IDENTITY_ENV_VAR = '{{identityProviders.[0].envVarName}}';

async function getApiKey(): Promise<string> {
  if (process.env.LOCAL_DEV === '1') {
    const apiKey = process.env[IDENTITY_ENV_VAR];
    if (!apiKey) {
      throw new Error(`${IDENTITY_ENV_VAR} not found. Add ${IDENTITY_ENV_VAR}=your-key to .env.local`);
    }
    return apiKey;
  }
  return withApiKey({ providerName: IDENTITY_PROVIDER_NAME }, async (apiKey: string) => apiKey)();
}

export function loadModel(): GoogleModel {
  return new GoogleModel({
    clientArgs: { apiKey: getApiKey },
    modelId: 'gemini-2.5-flash',
  });
}
{{/if}}
